### PR TITLE
Capture logback logger context properties

### PR DESCRIPTION
**Pull request by laurit**
Related to https://github.com/open-telemetry/opentelemetry-java-instrumentation/issues/9543

### CODE REVIEW

File: instrumentation/logback/logback-appender-1.0/library/src/main/java/io/opentelemetry/instrumentation/logback/appender/v1_0/internal/LoggingEventMapper.java [hunk 1]
```diff
@@ -46,18 +47,21 @@ public final class LoggingEventMapper {
...
+  private final boolean captureLoggerContext;
 
   public LoggingEventMapper(
       boolean captureExperimentalAttributes,
       List<String> captureMdcAttributes,
       boolean captureCodeAttributes,
       boolean captureMarkerAttribute,
-      boolean captureKeyValuePairAttributes) {
+      boolean captureKeyValuePairAttributes,
+      boolean captureLoggerContext) {
```
Comment (breedx-splk) [inline instrumentation/logback/logback-appender-1.0/library/src/main/java/io/opentelemetry/instrumentation/logback/appender/v1_0/internal/LoggingEventMapper.java:58]: This is a pretty sizable argument count -- and all those consecutive booleans are pretty prone to error. If nothing else, it's impossible for a reader to really grok.  A builder could help. 🤷🏻

File: instrumentation/logback/logback-appender-1.0/library/src/test/java/io/opentelemetry/instrumentation/logback/appender/v1_0/internal/LoggingEventMapperTest.java [hunk 2]
```diff
@@ -57,7 +58,7 @@ void testSome() {
   void testAll() {
     // given
     LoggingEventMapper mapper =
-        new LoggingEventMapper(false, singletonList("*"), false, false, false);
+        new LoggingEventMapper(false, singletonList("*"), false, false, false, false);
```
Comment (breedx-splk) [inline instrumentation/logback/logback-appender-1.0/library/src/test/java/io/opentelemetry/instrumentation/logback/appender/v1_0/internal/LoggingEventMapperTest.java:61]: Are we missing test coverage for the new `true` case?

**Review by breedx-splk — APPROVED** [review]
Looks like a step forward.

File: instrumentation/logback/logback-appender-1.0/library/src/test/java/io/opentelemetry/instrumentation/logback/appender/v1_0/internal/LoggingEventMapperTest.java (hunk 2 above)
Comment (laurit) [reply instrumentation/logback/logback-appender-1.0/library/src/test/java/io/opentelemetry/instrumentation/logback/appender/v1_0/internal/LoggingEventMapperTest.java:61]: It is tested in `OpenTelemetryAppenderTest`

File: instrumentation/logback/logback-appender-1.0/library/src/main/java/io/opentelemetry/instrumentation/logback/appender/v1_0/internal/LoggingEventMapper.java (hunk 1 above)
Comment (laurit) [reply instrumentation/logback/logback-appender-1.0/library/src/main/java/io/opentelemetry/instrumentation/logback/appender/v1_0/internal/LoggingEventMapper.java:58]: created https://github.com/open-telemetry/opentelemetry-java-instrumentation/issues/9889

**Pull request merged**